pkg/bzz/address.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bzz
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/swarm"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var ErrInvalidAddress = errors.New("invalid address")
+
+type Address struct {
+	Underlay  ma.Multiaddr
+	Overlay   swarm.Address
+	Signature []byte
+}
+
+type addressJSON struct {
+	Overlay   string
+	Underlay  string
+	Signature string
+}
+
+func NewAddress(signer crypto.Signer, underlay ma.Multiaddr, overlay swarm.Address, networkID uint64) (*Address, error) {
+	networkIDBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(networkIDBytes, networkID)
+	signature, err := signer.Sign(append(underlay.Bytes(), networkIDBytes...))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Address{
+		Underlay:  underlay,
+		Overlay:   overlay,
+		Signature: signature,
+	}, nil
+}
+
+func Parse(underlay, overlay, signature []byte, networkID uint64) (*Address, error) {
```

Comment:
if package would be called `bzzaddr` i think it would make the usage of such a function more clear in terms of outcome

Reply:
Or to call the function `ParseAddress`?